Newer pytz breaks khal

### Steps To Reproduce
Steps to reproduce the behavior:
1. build `khal` from `master`
2. git version error on pytz

### Build log
```console
$ nix log /nix/store/0nr02zra1dmz3j4a57ipgv4zh7azjrli-khal-0.10.5.drv                                                                                                                                
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
Sourcing pytest-check-hook
Using pytestCheckPhase
Removing setuptoolsCheckPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/w4ys3w7aia80xkcwwnzplsw3smm7sdk4-khal-0.10.5.tar.gz
source root is khal-0.10.5
setting SOURCE_DATE_EPOCH to timestamp 1656250614 of file khal-0.10.5/setup.cfg
@nix { "action": "setPhase", "phase": "patchPhase" }
...skipping...
tests/vdir_test.py .                                                     [ 96%]
tests/vtimezone_test.py ..E                                              [ 97%]
tests/ui/test_calendarwidget.py ...                                      [ 98%]
tests/ui/test_editor.py ....                                             [ 99%]
tests/ui/test_widgets.py .                                               [100%]

==================================== ERRORS ====================================
_____________________ ERROR at setup of test_event_no_dst ______________________

    @pytest.fixture(scope='session')
    def pytz_version():
        """Return the version of pytz as a tuple."""
>       year, month = pytz.__version__.split('.')
E       ValueError: too many values to unpack (expected 2)

tests/conftest.py:112: ValueError
________________________ ERROR at setup of test_bogota _________________________

    @pytest.fixture(scope='session')
    def pytz_version():
        """Return the version of pytz as a tuple."""
>       year, month = pytz.__version__.split('.')
E       ValueError: too many values to unpack (expected 2)

tests/conftest.py:112: ValueError
=========================== short test summary info ============================
ERROR tests/event_test.py::test_event_no_dst - ValueError: too many values to...
ERROR tests/vtimezone_test.py::test_bogota - ValueError: too many values to u...
====== 302 passed, 3 deselected, 1 xfailed, 2 xpassed, 2 errors in 8.38s =======
lines 277-300/300 (END)
```

### Additional context

Fixed upstream, but not released https://github.com/pimutils/khal/issues/1180

### Notify maintainers
@gebner 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.19.11, NixOS, 22.11 (Raccoon), 22.11.20220929.dirty`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(toastal): `""`
```

--- a/pkgs/applications/misc/khal/default.nix
+++ b/pkgs/applications/misc/khal/default.nix
@@ -1,32 +1,71 @@
-{ lib, stdenv, pkgs, python3, fetchpatch, glibcLocales, installShellFiles }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, glibcLocales
+, installShellFiles
+, python3
+}:
 
-with python3.pkgs; buildPythonApplication rec {
+python3.pkgs.buildPythonApplication rec {
   pname = "khal";
   version = "0.10.5";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-Tu+3rDAqJthgbbOSgXWHpO2UwnoVvy6iEWFKRk/PDHY=";
+  src = fetchFromGitHub {
+    owner = "pimutils";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-FneJmoAOb7WjSgluCwlspf27IU3MsQZFKryL9OSSsUw=";
   };
 
-  propagatedBuildInputs = [
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  nativeBuildInputs = [
+    glibcLocales
+    installShellFiles
+  ] ++ (with python3.pkgs; [
+    setuptools-scm
+    sphinx
+    sphinxcontrib_newsfeed
+  ]);
+
+  propagatedBuildInputs = with python3.pkgs;[
     atomicwrites
     click
     click-log
     configobj
-    python-dateutil
+    freezegun
     icalendar
     lxml
+    pkginfo
     pkgs.vdirsyncer
+    python-dateutil
     pytz
     pyxdg
     requests-toolbelt
     tzlocal
     urwid
-    pkginfo
-    freezegun
   ];
-  nativeBuildInputs = [ setuptools-scm sphinx sphinxcontrib_newsfeed installShellFiles ];
+
+  checkInputs = with python3.pkgs;[
+    freezegun
+    hypothesis
+    packaging
+    pytestCheckHook
+    vdirsyncer
+  ];
+
+  patches = [
+    # Tests working with latest pytz version, https://github.com/pimutils/khal/pull/1183
+    (fetchpatch {
+      name = "support-later-pytz.patch";
+      url = "https://github.com/pimutils/khal/commit/53eb8a7426d5c09478c78d809c4df4391438e246.patch";
+      sha256 = "sha256-drGtvJlQ3qFUdeukRWCFycPSZGWG/FSRqnbwJzFKITc=";
+      excludes = [
+        "CHANGELOG.rst"
+      ];
+    })
+  ];
 
   postInstall = ''
     # shell completions
@@ -38,18 +77,13 @@ with python3.pkgs; buildPythonApplication rec {
     # man page
     PATH="${python3.withPackages (ps: with ps; [ sphinx sphinxcontrib_newsfeed ])}/bin:$PATH" \
     make -C doc man
-    install -Dm755 doc/build/man/khal.1 -t $out/share/man/man1
+    installManPage doc/build/man/khal.1
 
-    # desktop
+    # .desktop file
     install -Dm755 misc/khal.desktop -t $out/share/applications
   '';
 
   doCheck = !stdenv.isAarch64;
-
-  checkInputs = [
-    glibcLocales
-    pytestCheckHook
-  ];
 
   LC_ALL = "en_US.UTF-8";
 
@@ -59,10 +93,10 @@ with python3.pkgs; buildPythonApplication rec {
   ];
 
   meta = with lib; {
-    broken = stdenv.isDarwin;
-    homepage = "http://lostpackets.de/khal/";
     description = "CLI calendar application";
+    homepage = "http://lostpackets.de/khal/";
     license = licenses.mit;
     maintainers = with maintainers; [ gebner ];
+    broken = stdenv.isDarwin;
   };
 }

